opmon/monitoring.py

```diff
@@ -375,6 +421,28 @@ def validate(self) -> None:
             first_run=True,
         )
         print(f"Dry run metrics SQL for {self.normalized_slug}")
+
+        if callable(self.before_execute_callback):
+            # Before and after are the same query for metrics: there's no
+            # modification preparing for dry run.
+            self.before_execute_callback(
+                metrics_sql,
+                None,
+                annotations={
+                    "slug": self.slug,
+                    "type": "validate_metrics_query_before",
+                    "submission_date": self.config.project.start_date,
+                },
+            )
+            self.before_execute_callback(
+                metrics_sql,
+                None,
+                annotations={
+                    "slug": self.slug,
+                    "type": "validate_metrics_query_after",
```

Comment:
```suggestion
                    "type": "validate_metrics_query_dryrun",
```

Same for the other queries?